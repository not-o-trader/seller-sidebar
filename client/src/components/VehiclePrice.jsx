import React from 'react';
import styled from 'styled-components';
import { formatMoney } from 'accounting';

const Wrapper = styled.div`
  width: 95%;
  margin: 0 auto;
`;

const FinancingLink = styled.a`
  margin-right: 10px;
`;

const PaymentsLink = styled.a`
  float: right;
`;

const KBB = styled.div`
  margin-top: -8px;
  float: left;
`;

const FMR = styled.div`
  margin: auto 0;
`;

class VehiclePrice extends React.Component {
  constructor(props) {
    super(props);
    this.handleCalculatorClick = this.handleCalculatorClick.bind(this);
  }

  handleCalculatorClick(e) {
    e.preventDefault();
    this.props.toggleModal();
  }

  render() {
    return (
      <Wrapper>
        <h3>{formatMoney(this.props.price)}</h3>
        <div>
          <div>
            <KBB>
              <img src="KBB_Badge.png" />
            </KBB>
            <FMR>
              <a href='#'>Is this price in the Fair Market Range?</a>
            </FMR>
          </div>
          <div>
            <p>
              <FinancingLink href='#'>$ Arrange Financing <span className="fas fa-external-link-alt"></span></FinancingLink>
              <PaymentsLink
                href='#'
                id="show-payment-calculator"
                onClick={this.handleCalculatorClick}
              >
                <span className="fas fa-calculator"></span> Calculate payment
              </PaymentsLink>
            </p>
          </div>
        </div>
      </Wrapper>
    );
  }
}

export default VehiclePrice;
