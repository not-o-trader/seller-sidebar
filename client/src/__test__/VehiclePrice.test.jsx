import React from 'react';
import { shallow } from 'enzyme';

import VehiclePrice from '../components/VehiclePrice.jsx';

describe('<VehiclePrice />', () => {
  it('renders', () => {
    const wrapper = shallow(
      <VehiclePrice />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
