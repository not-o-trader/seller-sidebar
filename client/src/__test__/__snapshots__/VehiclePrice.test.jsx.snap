// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehiclePrice /> renders 1`] = `
<styled.div>
  <h3>
    $0.00
  </h3>
  <div>
    <div>
      <styled.div>
        <img
          src="KBB_Badge.png"
        />
      </styled.div>
      <styled.div>
        <a
          href="#"
        >
          Is this price in the Fair Market Range?
        </a>
      </styled.div>
    </div>
    <div>
      <p>
        <styled.a
          href="#"
        >
          $ Arrange Financing 
          <span
            className="fas fa-external-link-alt"
          />
        </styled.a>
        <styled.a
          href="#"
          id="show-payment-calculator"
          onClick={[Function]}
        >
          <span
            className="fas fa-calculator"
          />
           Calculate payment
        </styled.a>
      </p>
    </div>
  </div>
</styled.div>
`;
